static/app/views/settings/organizationIntegrations/integrationReposAddRepository.tsx

```diff
@@ -126,22 +126,30 @@ export function IntegrationReposAddRepository({
 
   return (
     <DropdownWrapper>
-      <DropdownAutoComplete
-        items={dropdownItems}
-        onSelect={addRepo}
-        onChange={e => setSearch(e.target.value)}
-        emptyMessage={t('Please enter a repository name')}
-        noResultsMessage={t('No repositories found')}
+      <CompactSelect
+        size="xs"
+        menuWidth={250}
+        options={dropdownItems}
+        onChange={addRepo}
+        disabled={false}
+        menuTitle={t('Repositories')}
+        triggerLabel={t('Add Repository')}
+        emptyMessage={
+          query.isFetching
+            ? t('Searching\u2026')
+            : debouncedSearch
+              ? t('No repositories found')
+              : t('Please enter a repository name')
```

Comment:
two things here:

1) we need to combine `emptyMessage` and `noResultsMessage` from `DropdownAutoComplete` into a ternary here because `CompactSelect` doesn’t distinguish between those
2) we need to set the `Searching...` text manually too - `DropdownAutoComplete` was also doing this internally